### PR TITLE
Don't recommend saving shell completions to disk

### PR DESCRIPTION
**Pull request by wfchandler**
Currently our docs advise users of several shells to save the output of `oxide completion` to a file and source that in their init. This has led to users having outdated completions when they upgrade to a new version of the CLI.

Update the docs to recommend generating the completions on the fly at startup. Doing this will add a bit of latency to their shell startup, but user instructions should prioritize reliability over performance.

### CODE REVIEW

**Comment by wfchandler** [comment]
Validated updated instructions with `zsh`, `fish`, and `elvish` locally.

**Review by david-crespo — APPROVED** [review]
I'm for it.

image

**Comment by wfchandler** [comment]
@david-crespo it's a bit slower than that, since the shell needs to evaluate the generated text:

```sh
$ hyperfine  'zsh -c eval "$(oxide completion -s zsh)"'
Benchmark 1: zsh -c eval "$(oxide completion -s zsh)"
  Time (mean ± σ):      32.7 ms ±   0.7 ms    [User: 12.5 ms, System: 5.0 ms]
  Range (min … max):    31.6 ms …  35.4 ms    85 runs
```

**Comment by david-crespo** [comment]
Eagerly awaiting the M4 MBPs

image

**Review by dwradcliffe — APPROVED** [review]

File: cli/src/cmd_completion.rs [hunk 1]
```diff
@@ -27,26 +27,20 @@ use std::io;
 ///
```
Comment (ahl) [inline cli/src/cmd_completion.rs:27]: I don't get why this isn't working for me on macos:

```console
$ eval $(./target/debug/oxide completion --shell bash)
-bash: syntax error near unexpected token `)'
$ ./target/debug/oxide completion --shell bash > foo
$ . foo
# works fine
```

File: cli/src/cmd_completion.rs [hunk 2]
```diff
@@ -27,26 +27,20 @@ use std::io;
...
-/// `$fpath`, for example:
-///
-/// ```sh
-/// oxide completion -s zsh > ~/.zfunc/_oxide
-/// ```
-///
-/// and check that you have the following lines in your `~/.zshrc`:
+/// Add this to your `~/.zshrc`:
 ///
 /// ```sh
```
Comment (ahl) [inline cli/src/cmd_completion.rs:32]: these ```sh are in the console output for `oxide completion --help` -- we don't have to change that now, but that seems kind of lousy to me -- what do you think?

**Review by ahl — APPROVED** [review]

File: cli/src/cmd_completion.rs (hunk 1 above)
Comment (david-crespo) [reply cli/src/cmd_completion.rs:27]: try quotes around the `$()`?

Edit: yep this makes it not error out

Comment (david-crespo) [reply cli/src/cmd_completion.rs:27]: also the default shell on mac is zsh

File: cli/src/cmd_completion.rs (hunk 2 above)
Comment (david-crespo) [reply cli/src/cmd_completion.rs:32]: If we wanted to be really cool we could render the markdown

Comment (ahl) [reply cli/src/cmd_completion.rs:32]: https://github.com/clap-rs/clap/issues/2389

File: cli/src/cmd_completion.rs (hunk 1 above)
Comment (wfchandler) [reply cli/src/cmd_completion.rs:27]: Confirmed that error is due to lack of quotes.

```sh
bash-5.2$ eval $(./target/debug/oxide completion -s bash)
bash: syntax error near unexpected token `)'
bash-5.2$ eval "$(./target/debug/oxide completion -s bash)"
bash-5.2$ echo $?
0
```

Comment (ahl) [reply cli/src/cmd_completion.rs:27]: Nice! my bad.

**Pull request merged**